File: JihyeonChoi/BOJ/카잉 달력.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        for (int i = 0; i < T; i++) {
            int M = sc.nextInt();
            int N = sc.nextInt();
            int x = sc.nextInt();
            int y = sc.nextInt();

            boolean found = false;
            int year = x;

            while (year <= M * N) {
                if ((year - 1) % N + 1 == y) {
                    found = true;
                    System.out.println(year);
                    break;
                }
                year += M;
            }

            if (!found) {
                System.out.println(-1);
            }
        }
    }
}

File: JihyeonChoi/BOJ/카잉 달력.py
import sys
# 데이터의 입력을 나태내는 정수 T
T = int(sys.stdin.readline())

for i in range(T):
    M, N, x, y = map(int, sys.stdin.readline().split())
    
    #x를 기준으로 맞춰보자
    year = x
    found = False
    while year <= M*N:
        if (year-1)%N+1 == y:
            found=True
            print(year)
            break
        year+=M

    if not found:
        print(-1)




